components/ws-proxy/pkg/proxy/infoprovider.go

```diff
@@ -296,19 +318,23 @@ type workspaceInfoCache struct {
 	// WorkspaceCoords indexed by public (proxy) port (string)
 	coordsByPublicPort map[string]*WorkspaceCoords
 
-	mu sync.RWMutex
+	mu   *sync.RWMutex
+	cond *sync.Cond
```

Comment:
A comment on what the `cond` signal means. Here it's used as "some new info available" if I'm not mistaken (as opposed to "data has changed (added/removed)" as one might tempted to think).

Reply:
:+1: